### PR TITLE
do not depend on vault on public CI

### PR DESCRIPTION
**Pull request by Muen342**
Do not depend on vault on public CI

### Committer Checklist (excluded from commit message)
- [ x] Verify design and implementation 
- [ x] Verify test coverage and CI build status
- [ x] Verify documentation (including upgrade notes)

### CODE REVIEW

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -43,6 +43,12 @@ global_job_config:
       - echo $SEMAPHORE_WORKFLOW_ID
       - echo $SEMAPHORE_GIT_REPO_SLUG
       - checkout
+      - |
+        if [[ "${SEMAPHORE_ORGANIZATION_URL}" != *".semaphoreci.com" ]]; then
+          . vault-setup
+          . vault-sem-get-secret ci-reporting
+          . vault-sem-get-secret sox-semaphore-build-info
```
Comment (cchristous) [inline .semaphore/semaphore.yml:50]: Are you sure you are using these secrets? If so, this is fine, but at the least, I don't think this is a SOX repo, so I don't think the `sox-semaphore-build-info` is needed. It looks like you are using Testbreak below with `emit-ci-metrics -p -a test-results`. That consumes from `ci-reporting`.

Comment (Muen342) [reply .semaphore/semaphore.yml:50]: let me check

Comment (Muen342) [reply .semaphore/semaphore.yml]: yeah, I just tested it without it and it's still able to get the secret

**Review by cchristous — APPROVED** [review]

File: .semaphore/semaphore.yml [hunk 2]
```diff
@@ -43,6 +43,10 @@ global_job_config:
       - echo $SEMAPHORE_WORKFLOW_ID
       - echo $SEMAPHORE_GIT_REPO_SLUG
       - checkout
+      - |
+        if [[ "${SEMAPHORE_ORGANIZATION_URL}" != *".semaphoreci.com" ]]; then
+          . vault-setup
```
Comment (cchristous) [inline .semaphore/semaphore.yml:48]: You could move this call to the block below to removal a conditional, but this is fine

**Review by dajac — APPROVED** [review]
LGTM

**Pull request merged**